pkg/operation/etcdencryption/encryptionconfiguration.go

```diff
@@ -0,0 +1,178 @@
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package encryptionconfiguration
+
+import (
+	"fmt"
+
+	v1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/runtime/serializer"
+	"k8s.io/apimachinery/pkg/runtime/serializer/json"
+	utilruntime "k8s.io/apimachinery/pkg/util/runtime"
+	"k8s.io/apimachinery/pkg/util/sets"
+	apiserverconfigv1 "k8s.io/apiserver/pkg/apis/config/v1"
+)
+
+// NewPassiveConfiguration creates a initial configuration for etcd encryption
+// The list of encryption providers contains identity as first provider, which has
+// the effect, that this configuration does not yet encrypt written secrets. The
+// configuration has to be activated to actually encrypt written secrets.
+// Nevertheless, an encryption provider aescbc is already contained in the configuration
+// at the second position in the list of providers. A key is created for aescbc with
+// the key's name containing the timestamp when it was created in UTC.
+//
+// apiVersion: apiserver.config.k8s.io/v1
+// kind: EncryptionConfiguration
+// resources:
+// - providers:
+//   - aescbc:
+//       keys:
+//       - name: key1559747207815249000
+//         secret: Y8LEzbtK/2mdXrw6W/faAxNLu+mTCmcQeWojShAJGEg=
+//   - identity: {}
+//   resources:
+//     metadata:
+//   - secrets
+func NewPassiveConfiguration() (*apiserverconfigv1.EncryptionConfiguration, error) {
+	key, err := NewEncryptionKey()
+	if err != nil {
+		return nil, err
+	}
+	return &apiserverconfigv1.EncryptionConfiguration{
+		TypeMeta: v1.TypeMeta{
+			Kind:       ecKind,
+			APIVersion: ecAPIVersion,
+		},
+		Resources: []apiserverconfigv1.ResourceConfiguration{
+			{
+				Resources: ecEncryptedResources,
+				Providers: []apiserverconfigv1.ProviderConfiguration{
+					{Identity: &apiserverconfigv1.IdentityConfiguration{}},
+					{AESCBC: &apiserverconfigv1.AESConfiguration{
+						Keys: []apiserverconfigv1.Key{
+							*key,
+						},
+					}},
+				},
+			},
+		},
+	}, nil
+}
+
+// LoadFromYAML loads an EncryptionConfiguration from a YAML String as Byte Array.
+func LoadFromYAML(ecYamlBytes []byte) (*apiserverconfigv1.EncryptionConfiguration, error) {
+	scheme := runtime.NewScheme()
+	codecs := serializer.NewCodecFactory(scheme)
+	utilruntime.Must(apiserverconfigv1.AddToScheme(scheme))
+	ec := &apiserverconfigv1.EncryptionConfiguration{}
+	_, _, err := codecs.UniversalDecoder().Decode(ecYamlBytes, nil, ec)
+	if err != nil {
+		return nil, fmt.Errorf("error while decoding EncryptionConfiguration from yamlArray: %v", err)
+	}
+	return ec, nil
+}
+
+// ConvertToYAML converts an EncryptionConfiguration to a YAML representation.
+func ConvertToYAML(ec *apiserverconfigv1.EncryptionConfiguration) ([]byte, error) {
+	scheme := runtime.NewScheme()
```

Comment:
Once this is in `init`, you can reuse the codec here.

Reply:
created init function and moved parts there